arangod/MMFiles/MMFilesWalRecoverState.cpp

```diff
@@ -947,19 +947,17 @@ bool MMFilesWalRecoverState::ReplayMarker(MMFilesMarker const* marker,
 
           return state->canContinue();
         } else {
-          auto ctx = transaction::StandaloneContext::Create(*vocbase);
-          arangodb::SingleCollectionTransaction trx(
-            ctx, *col, AccessMode::Type::WRITE
-          );
-          std::shared_ptr<arangodb::Index> unused;
-          int res = physical->restoreIndex(&trx, payloadSlice, unused);
-
-          if (res != TRI_ERROR_NO_ERROR) {
+          
+          try {
+            bool created;
+            std::shared_ptr<arangodb::Index> unused;
+            unused = physical->createIndex(payloadSlice, /*restore*/true, created);
```

Comment:
declaration of and assignment to variable `unused` can be done in a single statement.